[PDI-15689] Log level set for schedules do not preserve after doing a RESTORE using the import-export utility

AliaksandrShuhayeu opened the pull request:
[PDI-15689] Log level set for schedules do not preserve after doing a RESTORE using the import-export utility

- fixed problem restoring job parameters as pdi parameters
- added backup pdi parameters
- added restoring pdi parameters

@mdamour1976 Could you please review and merge it?
My changes improved unit test coverage in my changes files.

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
ant -Dtestreports.dir=bin/reports/unit-test -f extensions/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f extensions/build.xml jacoco-integration checkstyle publish-local
```

#### Cleanup Commands
```sh
rm -r ~/.ivy2/local || echo no publish local to remove
```

#### Changed files

```console
  extensions/src/org/pentaho/platform/plugin/services/exporter/ScheduleExportUtil.java
  extensions/src/org/pentaho/platform/web/http/api/resources/JobScheduleRequest.java
  extensions/src/org/pentaho/platform/web/http/api/resources/SchedulerResourceUtil.java
  extensions/src/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java
  extensions/test-src/org/pentaho/platform/plugin/services/exporter/ScheduleExportUtilTest.java
  extensions/test-src/org/pentaho/platform/web/http/api/resources/SchedulerResourceUtilTest.java
  extensions/test-src/org/pentaho/platform/web/http/api/resources/services/SchedulerServiceTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.platform.config.MondrianConfigProperties
 * Branch Change:  + 7.14%
 * Instruction Change:  + 3.16%

###### org.pentaho.platform.plugin.services.exporter.ScheduleExportUtil
 * Branch Change:  + .43%
 * Complexity Change:  + .72%
 * Instruction Change:  + .09%
 * Line Change:  + .03%

###### org.pentaho.platform.plugin.services.metadata.PentahoMetadataDomainRepository
 * Branch Change: **-.57%**:small_red_triangle_down:
 * Instruction Change: **-.86%**:small_red_triangle_down:
 * Line Change: **-.76%**:small_red_triangle_down:

###### org.pentaho.platform.plugin.services.metadata.PentahoMetadataInformationMap
 * Branch Change:  + 6.25%
 * Complexity Change:  + 5.00%

###### org.pentaho.platform.web.http.api.resources.JobScheduleRequest
 * Complexity Change:  + 1.04%
 * Instruction Change:  + 1.70%
 * Line Change:  + 1.33%
 * Method Change:  + .59%

###### org.pentaho.platform.web.http.api.resources.SchedulerResourceUtil
 * Branch Change:  + .96%
 * Complexity Change:  + 1.62%
 * Instruction Change:  + 1.35%
 * Line Change:  + .82%

###### org.pentaho.platform.web.http.api.resources.services.SchedulerService
 * Instruction Change:  + .09%

#### Integration test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.platform.plugin.action.chartbeans.ChartAction
 * Instruction Change: **-.22%**:small_red_triangle_down:

###### org.pentaho.platform.web.http.api.resources.SchedulerResourceUtil
 * Branch Change:  + .35%
 * Complexity Change:  + .21%
 * Instruction Change:  + .23%
 * Line Change:  + .22%

###### org.pentaho.platform.web.http.api.resources.services.SchedulerService
 * Instruction Change: **-.05%**:small_red_triangle_down:

#### License header violations

> :heavy_exclamation_mark: Couldn't find acceptable copyright on extensions/test-src/org/pentaho/platform/plugin/services/exporter/ScheduleExportUtilTest.java

> :heavy_exclamation_mark: Couldn't find acceptable license on extensions/test-src/org/pentaho/platform/plugin/services/exporter/ScheduleExportUtilTest.java

#### Checkstyle violations

The code submitted is not complying with one or more of our guidelines. To perform these checks locally, [please use this tool](https://github.com/pentaho/pentaho-coding-standards).

###### /home/buildguy/project/extensions/test-src/org/pentaho/platform/plugin/services/exporter/ScheduleExportUtilTest.java

> :heavy_exclamation_mark: File does not end with a newline. at Line: 0

> :heavy_exclamation_mark: '||' should be on a new line. at Line: 257 Column: 59

> :heavy_exclamation_mark: '||' should be on a new line. at Line: 258 Column: 49

wingman-pentaho commented:
### Build Completed

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
ant -Dtestreports.dir=bin/reports/unit-test -f extensions/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f extensions/build.xml jacoco-integration checkstyle publish-local
```

#### Cleanup Commands
```sh
rm -r ~/.ivy2/local || echo no publish local to remove
```

#### Changed files

```console
  extensions/src/org/pentaho/platform/plugin/services/exporter/ScheduleExportUtil.java
  extensions/src/org/pentaho/platform/web/http/api/resources/JobScheduleRequest.java
  extensions/src/org/pentaho/platform/web/http/api/resources/SchedulerResourceUtil.java
  extensions/src/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java
  extensions/test-src/org/pentaho/platform/plugin/services/exporter/ScheduleExportUtilTest.java
  extensions/test-src/org/pentaho/platform/web/http/api/resources/SchedulerResourceUtilTest.java
  extensions/test-src/org/pentaho/platform/web/http/api/resources/services/SchedulerServiceTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.platform.plugin.services.exporter.ScheduleExportUtil
 * Branch Change:  + .43%
 * Complexity Change:  + .72%
 * Instruction Change:  + .09%
 * Line Change:  + .03%

###### org.pentaho.platform.plugin.services.metadata.PentahoMetadataDomainRepository
 * Branch Change: **-.57%**:small_red_triangle_down:
 * Complexity Change: **-.73%**:small_red_triangle_down:

###### org.pentaho.platform.plugin.services.metadata.PentahoMetadataInformationMap
 * Branch Change: **-6.25%**:small_red_triangle_down:
 * Complexity Change: **-5.00%**:small_red_triangle_down:

###### org.pentaho.platform.web.http.api.resources.JobScheduleRequest
 * Complexity Change:  + 1.04%
 * Instruction Change:  + 1.70%
 * Line Change:  + 1.33%
 * Method Change:  + .59%

###### org.pentaho.platform.web.http.api.resources.SchedulerResourceUtil
 * Branch Change:  + .96%
 * Complexity Change:  + 1.62%
 * Instruction Change:  + 1.35%
 * Line Change:  + .82%

###### org.pentaho.platform.web.http.api.resources.services.SchedulerService
 * Instruction Change:  + .09%

#### Integration test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.platform.web.http.api.resources.SchedulerResourceUtil
 * Branch Change:  + .35%
 * Complexity Change:  + .21%
 * Instruction Change:  + .23%
 * Line Change:  + .22%

###### org.pentaho.platform.web.http.api.resources.services.SchedulerService
 * Instruction Change: **-.05%**:small_red_triangle_down:

pamval commented:
@pedrofvteixeira @e-cuellar please review

AliaksandrShuhayeu commented:
[PDI-15689] Log level set for schedules do not preserve after doing a RESTORE using the import-export utility

- fixed problem restoring job parameters as pdi parameters
- added backup pdi parameters
- added restoring pdi parameters

@pedrofvteixeira @e-cuellar @mdamour1976 Could you please review and merge it?
My changes improved unit test coverage in my changes files.
